File: leetcode/1-two-sum/solution2.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function twoSum(nums, target) {
  const map = {};
  for (let i = 0; i < nums.length; i++) {
    if (map.hasOwnProperty(target - nums[i])) {
      return [i, map[target - nums[i]]];
    } else {
      map[nums[i]] = i;
    }
  }
}

File: leetcode/1-two-sum/solution2.java
import java.util.HashMap;

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */
class Solution {
  public int[] twoSum(int[] nums, int target) {
    int[] res = new int[2];
    HashMap hm = new HashMap<Integer, Integer>();
    for (int i = 0; i < nums.length; i++) {
      if (hm.containsKey(target - nums[i])) {
        res[0] = (int) hm.get(target - nums[i]);
        res[1] = i;
        return res;
      }
      hm.put(nums[i], i);
    }
    return res;
  }
}
